src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -64,20 +66,34 @@ export const useWorkspace = (namespace, name) : WorkspaceDetails => {
     console.assert(!!workspace, 'initialization should not be called before workspace details are fetched')
 
     if (isGoogleWorkspace(workspace)) {
-      !workspaceInitialized ? checkGooglePermissions(workspace) : loadGoogleBucketLocation(workspace)
+      !workspaceInitialized ? checkGooglePermissions(workspace) : loadGoogleBucketLocationIgnoringError(workspace)
     } else if (isAzureWorkspace(workspace)) {
       !workspaceInitialized ? checkAzureStorageExists(workspace) : loadAzureStorageDetails(workspace)
     }
   }
 
   const checkGooglePermissions = async workspace => {
     try {
+      // Because checkBucketReadAccess can succeed and subsequent calls to get the bucket location or storage
+      // cost estimate may fail (due to caching of previous failure results), do not consider permissions
+      // to be done syncing until all the methods that we know will be called quickly in succession succeed.
+      // This is not guaranteed to eliminate the issue, but it improves the odds.
       await Ajax(signal).Workspaces.workspace(namespace, name).checkBucketReadAccess()
+      if (Utils.canWrite(workspace.accessLevel)) {
+        // Calls done on the Workspace Dashboard. We could store the results and pass them
+        // through, but then we would have to do it checkWorkspaceInitialization as well,
+        // and nobody else actually needs these values.
+        await Ajax(signal).Workspaces.workspace(namespace, name).storageCostEstimate()
```

Comment:
In my testing, I have sometimes seen `storageCostEstimate` fail when `Dashboard` calls it, even after all these succeed here (in the case of workspaces being viewed for the first time). It's unfortunate, but I don't think it's worth pushing storage cost estimate and bucket usage state into `useWorkspace` and passing it around, given that the Dashboard is the only consumer of the information.